referenceframe/frame.go

```diff
@@ -218,7 +218,7 @@ func NewNamedFrame(frame Frame, name string) Frame {
 
 // NewStaticFrame creates a frame given a pose relative to its parent. The pose is fixed for all time.
 // Pose is not allowed to be nil.
-func NewStaticFrame(name string, pose spatial.Pose) (Frame, error) {
+func NewStaticFrame(name string, pose spatial.Pose) (*staticFrame, error) {
```

Comment:
Theres a couple functions that specifically require a static frame to work.  `ToLinkConfig` being public and accepting a private struct makes it so that it can't actually be used outside the referenceframe package.  This is my solution and follows the Go idiom of returning the object rather than the interface.

Reply:
Linter is complaining about this, calling it "annoying". I'm inclined to disagree and ignore but open to other ideas of how to do this

Reply:
> Go idiom of returning the object rather than the interface

This doesn't really apply when the struct it private. I'm inclined to agree with the linter; don't return a private struct from a public function.

`NewLinkConfig` also should not be a public function taking a private struct. Let's just fix that, have it take a `Frame` and then it can error if that frame has >0 dof.

Normally I'm very cautious of breaking changes to public functions but in this case it's guaranteed not to break anyone as `NewLinkConfig` is not usable outside `referenceframe`.

Reply:
Alright I'm fine with this path too